File: python/01-magic-receipt/test/receipt_test.py
from src.receipt import Receipt

def test_regular_customer():
    receipt = Receipt()
    assert receipt.calculate_total([10, 20, 30]) == 59.4

def test_member_customer():
    receipt = Receipt()
    assert receipt.calculate_total([10, 20, 30], "member") == 56.16

def test_vip_customer():
    receipt = Receipt()
    assert receipt.calculate_total([10, 20, 30], "vip") == 47.68

def test_bonus_discount_over_50():
    receipt = Receipt()
    assert receipt.calculate_total([60]) == 59.4

def test_vip_extra_discount():
    receipt = Receipt()
    assert receipt.calculate_total([100], "vip") == 84.4

def test_empty_items():
    receipt = Receipt()
    assert receipt.calculate_total([]) == 0


def test_exactly_50_no_bonus():
    receipt = Receipt()
    assert receipt.calculate_total([50]) == 54.0

File: ruby/01-magic-receipt/test/receipt_test.rb
# frozen_string_literal: true

require 'minitest/autorun'
require_relative '../src/receipt'

class ReceiptTest < Minitest::Test
  def setup
    @receipt = Receipt.new
  end

  def test_regular_customer
    assert_equal 59.4, @receipt.calculate_total([10, 20, 30])
  end

  def test_member_customer
    assert_equal 56.16, @receipt.calculate_total([10, 20, 30], 'member')
  end

  def test_vip_customer
    assert_equal 47.68, @receipt.calculate_total([10, 20, 30], 'vip')
  end

  def test_bonus_discount_over_50
    assert_equal 59.4, @receipt.calculate_total([60])
  end

  def test_vip_extra_discount
    assert_equal 84.4, @receipt.calculate_total([100], 'vip')
  end

  def test_empty_items
    assert_equal 0, @receipt.calculate_total([])
  end

  def test_exactly_50_no_bonus
    assert_equal 54.0, @receipt.calculate_total([50])
  end
end
